ui/components/multichain/create-account/create-account.tsx

```diff
@@ -20,19 +16,41 @@ import {
 import { MetaMetricsContext } from '../../../contexts/metametrics';
 import { Display } from '../../../helpers/constants/design-system';
 
-export const CreateAccount = ({ onActionComplete }) => {
+type CreateAccountOptions = {
+  /**
+   * Callback to get the next available account name.
+   */
+  getNextAvailableAccountName: (accounts: InternalAccount[]) => Promise<string>;
+
+  /**
+   * Callback to create the account.
+   */
+  onCreateAccount: (name: string) => Promise<void>;
+
+  /**
+   * Callback called once the account has been created
+   */
+  onActionComplete: (completed: boolean) => Promise<void>;
+};
+
+export const CreateAccount = ({
```

Comment:
What do you think about using `React.memo` to avoid unnecessary re-renderings?

https://react.dev/reference/react/memo

Reply:
seems like a good practice indeed 🙈 lemme try that.